DatadogWebViewTracking/Sources/WebViewTracking.swift

```diff
@@ -4,55 +4,126 @@
+
+    static func enable(tracking controller: WKUserContentController, hosts: Set<String>, hostsSanitizer: HostsSanitizing, in core: DatadogCoreProtocol) {
+        let isTracking = controller.userScripts.contains { $0.source.starts(with: Self.jsCodePrefix) }
+        guard !isTracking else {
+            DD.logger.warn("`startTrackingDatadogEvents(core:hosts:)` was called more than once for the same WebView. Second call will be ignored. Make sure you call it only once.")
+            return
+       }
+
+        let bridgeName = DDScriptMessageHandler.name
+
+        let messageHandler = DDScriptMessageHandler(
+            emitter: MessageEmitterCore(core: core)
+        )
+
+        controller.add(messageHandler, name: bridgeName)
+
+        // WebKit installs message handlers with the given name format below
+        // We inject a user script to forward `window.{bridgeName}` to WebKit's format
+        let webkitMethodName = "window.webkit.messageHandlers.\(bridgeName).postMessage"
+        // `WKScriptMessageHandlerWithReply` returns `Promise` and `browser-sdk` expects immediate values.
+        // We inject a user script to return `allowedWebViewHosts` instead of using `WKScriptMessageHandlerWithReply`
+        let sanitizedHosts = hostsSanitizer.sanitized(
+            hosts: hosts,
+            warningMessage: "The allowed WebView host configured for Datadog SDK is not valid"
+        )
+        let allowedWebViewHostsString = sanitizedHosts
+            .map { return "\"\($0)\"" }
+            .joined(separator: ",")
+
+        let js = """
+        \(Self.jsCodePrefix)
+        window.\(bridgeName) = {
+          send(msg) {
+            \(webkitMethodName)(msg)
+          },
+          getAllowedWebViewHosts() {
+            return '[\(allowedWebViewHostsString)]'
+          }
         }
+        """
+
+        controller.addUserScript(
+            WKUserScript(
+                source: js,
+                injectionTime: .atDocumentStart,
+                forMainFrameOnly: false
+            )
+        )
     }
+#endif
 }
 
-extension WebViewTracking {
-    /// Sends a bag of data to the message bus
-    /// - Parameter body: The data to send, it must be parsable to `WebViewTrackingMessage`
-    internal func send(body: Any) throws {
-        let message = try WebViewTrackingMessage(body: body)
-        try send(message: message)
+extension WebViewTracking: InternalExtended { }
+extension InternalExtension where ExtendedType: WebViewTracking {
+    /// Creates a web-view message emitter for cross-platform.
+    ///
+    /// Cross platform SDKs should instantiate a `MessageEmitter` implementation from
+    /// this method and pass WebView related messages using the message bus of the core.
+    ///
+    /// - Parameter core: The Datadog SDK core instance
+    /// - Returns: A `MessageEmitter` instance
+    public static func messageEmitter(in core: DatadogCoreProtocol) -> MessageEmitter {
```

Comment:
I'm introducing this interface for CP SDKs so they can transmit WebView messages. It requires `MessageEmitter` and `WebViewTrackingMessage` to be public.

Reply:
question/ Could `MessageEmitter` and `WebViewTrackingMessage` be nested under `InternalExtension` namespace? If so, let's do. Otherwise we pollute app's namespace with indirect and common types like `JSON` typealias.

Reply:
MessageEmitter is a protocol and cannot be nested, but it doesn't need 
to be a protocol.. Let me see how I can hide all of that 👍